src/Bridge/Symfony/Bundle/DataPersister/TraceableChainDataPersister.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DataPersister;
+
+use ApiPlatform\Core\DataPersister\ChainDataPersister;
+use ApiPlatform\Core\DataPersister\DataPersisterInterface;
+
+/**
+ * @author Anthony GRASSIOT <antograssiot@free.fr>
+ */
+final class TraceableChainDataPersister implements DataPersisterInterface
+{
+    private $persisters = [];
+    private $persistersResponse = [];
+    private $decorated;
+
+    public function __construct(DataPersisterInterface $dataPersister)
+    {
+        if ($dataPersister instanceof ChainDataPersister) {
+            $this->decorated = $dataPersister;
+            $this->persisters = $dataPersister->persisters;
+        }
+    }
+
+    public function getPersistersResponse(): array
+    {
+        return $this->persistersResponse;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($data): bool
+    {
+        return $this->decorated->supports($data);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function persist($data)
```

Comment:
The logic in this method and in `remove` looks duplicated. Can't you extract it in a private method, and just pass the name of the decorated method to call (`persist` or `remove`) as a parameter?

Reply:
Done

Reply:
I didn't needed to pass the method name as parameter as the return is different in both case, is it ok like that ?